### ISSUE TITLE
MEGA: segmentation fault in long-running process

### ISSUE THREAD

**Issue by mpldr** [rclone/rclone#6336]
#### The associated forum post URL from `https://forum.rclone.org`

not applicable

#### What is the problem you are having with rclone?

MEGA backend crashed after a long-running job (about 19 hours) due to a nil-pointer dereference in t3rm1n4l's library. Since that library does not seem to be actively maintained, I am therefore raising that bug here.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x55f29cc77c06]

goroutine 35 [running]:
github.com/t3rm1n4l/go-mega.(*Mega).processUpdateNode(0xc0003f8690, {0xc000d87b80, 0x149, 0x160})
	github.com/t3rm1n4l/go-mega@v0.0.0-20200416171014-ffad7fcb44b8/mega.go:1848 +0x126
github.com/t3rm1n4l/go-mega.(*Mega).pollEvents(0xc0003f8690)
	github.com/t3rm1n4l/go-mega@v0.0.0-20200416171014-ffad7fcb44b8/mega.go:2006 +0xf50
created by github.com/t3rm1n4l/go-mega.(*Mega).getFileSystem
	github.com/t3rm1n4l/go-mega@v0.0.0-20200416171014-ffad7fcb44b8/mega.go:943 +0x2f4
```

Appears to be different to #3140 because the MEGA drive was not touched while rclone was working

#### What is your rclone version (output from `rclone version`)

```
rclone v1.59.0
- os/version: arch (64 bit)
- os/kernel: 5.18.10-arch1-1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.18.3
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Arch Linux x64_86

#### Which cloud storage system are you using? (e.g. Google Drive)

MEGA

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone move mega:Archive gdrive:Backups/old_mega -P`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

Reproduction attempt still running, will update if it can be reliably reproduced.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
This looks easy to fix - give this a go

[v1.60.0-beta.6374.168c51c7c.fix-mega-nil-pointer](https://beta.rclone.org/branch/fix-mega-nil-pointer/v1.60.0-beta.6374.168c51c7c.fix-mega-nil-pointer/) on branch [fix-mega-nil-pointer](https://github.com/rclone/rclone/tree/fix-mega-nil-pointer) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.60 and v1.59.1
